### ISSUE TITLE
Bounding Box Issue

### ISSUE THREAD

**Issue by abeyang00** [AlexeyAB/darknet#739]
Originally, when i use "demo" option with this forked version, i get the result like

![image](https://user-images.githubusercontent.com/36392866/39572955-33c0d7c6-4f0c-11e8-935a-7860cd90cd89.png)

I have now changed the setting to ROS and when i try to detect traffic lights, i get the result like
![image](https://user-images.githubusercontent.com/36392866/39572967-3d92c048-4f0c-11e8-86da-a28156231773.png)

Do you have any idea what could be the issue? Will it have to do with nms?

**Comment by abeyang00** [comment]
Basically I'm trying to change the settings to ROS using https://github.com/leggedrobotics/darknet_ros this fork. I'm using same cfg, weight files and haven't changed anything and yes i'm using yolov3. I'm not getting the same result as when i used 'demo' option in your fork.

Almost everything seems to be the same but there's a sllight difference in draw_detections function

void draw_detections(image im, detection *dets, int num, float thresh, char **names, image **alphabet, int classes)
{
    int i,j;

    for(i = 0; i < num; ++i){
        char labelstr[4096] = {0};
        int class = -1;
        for(j = 0; j < classes; ++j){
            if (dets[i].prob[j] > thresh){
                if (class < 0) {
                    strcat(labelstr, names[j]);
                    class = j;
                } else {
                    strcat(labelstr, ", ");
                    strcat(labelstr, names[j]);
                }
                printf("%s: %.0f%%\n", names[j], dets[i].prob[j]*100);
            }
        }
        if(class >= 0){
            int width = im.h * .006;

            /*
               if(0){
               width = pow(prob, 1./2.)*10+1;
               alphabet = 0;
               }
             */

            //printf("%d %s: %.0f%%\n", i, names[class], prob*100);
            int offset = class*123457 % classes;
            float red = get_color(2,offset,classes);
            float green = get_color(1,offset,classes);
            float blue = get_color(0,offset,classes);
            float rgb[3];

            //width = prob*20+2;

            rgb[0] = red;
            rgb[1] = green;
            rgb[2] = blue;
            box b = dets[i].bbox;
            //printf("%f %f %f %f\n", b.x, b.y, b.w, b.h);

            int left  = (b.x-b.w/2.)*im.w;
            int right = (b.x+b.w/2.)*im.w;
            int top   = (b.y-b.h/2.)*im.h;
            int bot   = (b.y+b.h/2.)*im.h;

            if(left < 0) left = 0;
            if(right > im.w-1) right = im.w-1;
            if(top < 0) top = 0;
            if(bot > im.h-1) bot = im.h-1;

            draw_box_width(im, left, top, right, bot, width, red, green, blue);
            if (alphabet) {
                image label = get_label(alphabet, labelstr, (im.h*.03));
                draw_label(im, top + width, left, label, rgb);
                free_image(label);
            }
            if (dets[i].mask){
                image mask = float_to_image(14, 14, 1, dets[i].mask);
                image resized_mask = resize_image(mask, b.w*im.w, b.h*im.h);
                image tmask = threshold_image(resized_mask, .5);
                embed_image(tmask, im, left, top);
                free_image(mask);
                free_image(resized_mask);
                free_image(tmask);
            }
        }
    }
}

**Comment by abeyang00** [comment]
@cipri-tom 
In my case, (x,y) and width and height ratios are wrong....
Did the abve code also solve the ratio issue?

**Comment by AlexeyAB** [comment]
> I get the same result as when i used your forked version.
> I'm guessing there's some code problem when the other forked version used the original repository....

Yes, there is some issue in the https://github.com/leggedrobotics/darknet_ros
Try to ask this question here: https://github.com/leggedrobotics/darknet_ros/issues

**Comment by abeyang00** [comment]
@AlexeyAB 
Yeah i will do that. You are not by chance working on ROS porting, right?
